generator/tools/templates/react/src/hooks/useEffectHmr.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef, EffectCallback, DependencyList, useEffect } from "react";
+
+/**
+ *
+ * @param effect
+ * @param dependencies
+ * @description Hook to prevent running the useEffect beyond the first render which sometimes occurs in Vite where hot reload causes the useEffect to run several times despite an empty depedency
+ *
+ */
+export default function useEffectHmr(
+    effect: EffectCallback,
+    dependancies?: DependencyList
```

Comment:
If we're keeping this function, `dependancies` -> `dependencies`.
And the last word in the @description above is misspelled too.
I suggest we remove this, though, and stick with `useEffect`.